test/functional/api/v2/discovered_hosts_controller_test.rb

```diff
@@ -54,16 +69,42 @@ def test_delete_discovered_host
     assert_response :success
   end
 
+  def test_provision_host
+    FactoryGirl.create(:organization, :name => 'SomeOrg')
+    FactoryGirl.create(:location, :name => 'SomeLoc')
+    host = Host::Discovered.import_host(@facts)
+    hostgroup = setup_hostgroup(host)
+    put :update,
+        id: host.id,
+        discovered_host: {
+          hostgroup_id: hostgroup.id
+        }
+
+    assert_response :success
+
+    # Get the managed host instance from the DB
+    actual = Host.find(host.id)
+
+    assert_equal hostgroup.id, actual.hostgroup_id
+    assert_equal hostgroup.architecture_id, actual.architecture_id
+    assert_equal hostgroup.operatingsystem_id, actual.operatingsystem_id
+    assert_equal hostgroup.medium_id, actual.medium_id
+    assert_equal hostgroup.ptable_id, actual.ptable_id
+    assert_equal hostgroup.domain_id, actual.domain_id
+  end
```

Comment:
Mind to extend this with three more fields - environment, puppet and ca? We had similar issues in the past with these too. It's optional, I buy it just like that too :-)